### DISCUSSION TITLE
LiFePO4 48 V

### DISCUSSION THREAD

**Discussion by stanhausc** [Louisvdw/dbus-serialbattery#559] [New BMS requests]
Hello, @Louisvdw, I'm a student at [PTI](https://www.pti.org.br), we have several renewable energy projects, one of them is the communication between the LiFePO4 48 V battery and the Venus OS.
Here's the battery communication protocol
[Protocolo de comunicação LFP - MODBUS_ rev.06.pdf](https://github.com/Louisvdw/dbus-serialbattery/files/8098726/Protocolo.de.comunicacao.LFP.-.MODBUS_.rev.06.pdf).
I'll work on it over the next few weeks, and if it goes well, I'll share the code.

Thanks for your great work.

**Comment by emuman049** [comment]
Is the battery mentioned in the documentation a MasterBattery Upower UE-48Li100-3U? Any Progress so far?

**Comment by stanhausc** [comment]
> 
This is not the battery (the pdf has the specifications), so far I'm able to read the voltage, current, SOC, alarms, number of cells and individual cell voltage data.

**Comment by emuman049** [comment]
Okay I was Just thinking it is the same one, since 3U was mentioned in the PDF and Upower ist from Spain. What kind of BMS is this?

**Comment by stanhausc** [comment]
> Ok, eu estava pensando que é o mesmo, já que o 3U foi mencionado no PDF e o Upower é da Espanha. Que tipo de BMS é esse?

The information I have about the BMS is:

![image](https://user-images.githubusercontent.com/85182432/201184257-fa0d21f1-3fba-4279-837f-55169baf2788.png)

**Comment by cnidus** [comment]
+1 I have a stack do lifepower4 24v EG4 batteries I’d love to support with CerboGX.

**Comment by drmiar** [comment]
May I ask? What Type of BMS is integrated in the batteries UE-48Li100-3U? Is there a possibility to make a firmware update? Software?  Thanks a Lot!
